apis/privilegedaccessmanager/v1beta1/privilegedaccessmanagerentitlement_types.go

```diff
@@ -0,0 +1,315 @@
+	// +optional
+	ResourceID *string `json:"resourceID,omitempty"`
+
+	// Optional. Who can create grants using this entitlement. This list should
+	// contain at most one entry.
+	// +optional
+	EligibleUsers *[]AccessControlEntry `json:"eligibleUsers,omitempty"`
+
+	// Optional. The approvals needed before access are granted to a requester. No
+	// approvals are needed if this field is null.
+	// +optional
+	ApprovalWorkflow *ApprovalWorkflow `json:"approvalWorkflow,omitempty"`
+
+	// Required. The maximum amount of time that access is granted for a request.
+	// A requester can ask for a duration less than this, but never more.
+	// +required
+	MaxRequestDuration string `json:"maxRequestDuration,omitempty"`
+
+	// The access granted to a requester on successful approval.
+	// +required
+	PrivilegedAccess PrivilegedAccess `json:"privilegedAccess,omitempty"`
+
+	// Required. The manner in which the requester should provide a justification
+	// for requesting access.
+	// +required
+	RequesterJustificationConfig RequesterJustificationConfig `json:"requesterJustificationConfig,omitempty"`
+
+	// Optional. Additional email addresses to be notified based on actions taken.
+	// +optional
+	AdditionalNotificationTargets *AdditionalNotificationTargets `json:"additionalNotificationTargets,omitempty"`
+}
+
+// AccessControlEntry is used to control who can do some operation.
+type AccessControlEntry struct {
+	// Optional. Users who are allowed for the operation. Each entry should be a
+	// valid v1 IAM principal identifier. The format for these is documented at:
+	// https://cloud.google.com/iam/docs/principal-identifiers#v1
+	// +optional
+	Principals *[]string `json:"principals,omitempty"`
+}
+
+// Different types of approval workflows that can be used to gate privileged
+// access granting.
+type ApprovalWorkflow struct {
+	// An approval workflow where users designated as approvers review and act
+	// on the grants.
+	// +required
+	ManualApprovals ManualApprovals `json:"manualApprovals,omitempty"`
+}
+
+// A manual approval workflow where users who are designated as approvers
+// need to call the 'ApproveGrant'/'DenyGrant' APIs for a grant. The workflow
+// can consist of multiple serial steps where each step defines who can act as
+// approver in that step and how many of those users should approve before the
+// workflow moves to the next step.
+//
+// This can be used to create approval workflows such as:
+//
+// * Require an approval from any user in a group G.
+// * Require an approval from any k number of users from a Group G.
+// * Require an approval from any user in a group G and then from a user U.
+//
+// A single user might be part of the 'approvers' ACL for multiple steps in this
+// workflow, but they can only approve once and that approval is only considered
+// to satisfy the approval step at which it was granted.
+type ManualApprovals struct {
+	// Optional. Do the approvers need to provide a justification for their
```

Comment:
Suggest removing this question since this is user facing comment.

Reply:
Will rephrase it. This field description comes directly from the API: https://cloud.google.com/iam/docs/reference/pam/rest/v1beta/folders.locations.entitlements#Entitlement.ManualApprovals - do we have any guideline on what is a good field description and what is not?